src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -714,12 +714,7 @@ private void SearchDefaultNodeAutocompleteCandidates()
             var queries = new List<string>() { "String", "Number Slider", "Integer Slider", "Number", "Boolean", "Watch", "Watch 3D", "Python Script" };
             foreach (var query in queries)
             {
-                var nodeSearchElement = tempSearchViewModel.Model.Entries.FirstOrDefault(n => n.Name == query);
-                if(nodeSearchElement == null)
-                {
-                    continue;
-                }
-                var foundNode = tempSearchViewModel.MakeNodeSearchElementVM(nodeSearchElement);
+                var foundNode = tempSearchViewModel.Search(query).Where(n => n.Name.Equals(query)).FirstOrDefault();
```

Comment:
This is reverting the performance improvement we did or re-apply?

Reply:
This Dynamo loading improvement change was applied by @chubakueno (Luis Rivera from Design Automated Testing Group - https://github.com/DynamoDS/Dynamo/pull/15529 ), even if you see the comments in the loading improvement PR there were several regressions and the fix applied for fixing the regression didn't have logic with why the regressions were reported (there was no follow-up on my comments). 

Then we have the options:
- keeping this fix and continue working on understanding why the t-spline nodes are not indexed on time
- merge this changes and create a specific task for checking this improvement and the problem with indexing t-spline nodes.

Please let me know what you think.

Reply:
I feel this will affect performance. If we revert the previous change, all the queries will be checked even when a node match is found and SearchDefaultNodeAutocompleteCandidates()  is called on every node autocomplete trigger.

Reply:
Will check if i can debug the tsplines loading issue. The fix applied was conforming to prior logic ignoring non-matching nodes, there may be some race condition to be investigated causing this.